test/src/unit-ctx.cc

```diff
@@ -103,3 +104,58 @@ TEST_CASE("C++ API: Test context tags", "[cppapi][ctx-tags]") {
     }
   }
 }
+
+TEST_CASE(
+    "C API: Test context data protocol", "[capi][ctx][data-protocol][rest]") {
+  tiledb::test::VFSTestSetup vfs_test_setup{};
+  tiledb_ctx_t* ctx = vfs_test_setup.ctx_c;
+
+  tiledb_data_protocol_t data_protocol;
+
+  SECTION("tiledb:// URI") {
+    if (vfs_test_setup.is_rest()) {
+      auto rc = tiledb_ctx_get_data_protocol(
+          ctx, "tiledb://workspace/teamspace/array", &data_protocol);
+      REQUIRE(rc == TILEDB_OK);
+      if (vfs_test_setup.is_legacy_rest()) {
+        REQUIRE(data_protocol == TILEDB_DATA_PROTOCOL_v2);
+      } else {
+        REQUIRE(data_protocol == TILEDB_DATA_PROTOCOL_v3);
+      }
+    }
+  }
+
+  SECTION("non-tiledb:// URI") {
+    auto rc = tiledb_ctx_get_data_protocol(
+        ctx, "azure://bucket/array", &data_protocol);
+    REQUIRE(rc == TILEDB_OK);
+    REQUIRE(data_protocol == TILEDB_DATA_PROTOCOL_v2);
+  }
+}
+
+TEST_CASE(
+    "C++ API: Test context data protocol",
+    "[cppapi][ctx][data-protocol][rest]") {
+  tiledb::test::VFSTestSetup vfs_test_setup{};
+  tiledb::Context ctx{vfs_test_setup.ctx()};
+
+  SECTION("tiledb:// URI") {
+    if (vfs_test_setup.is_rest()) {
+      if (vfs_test_setup.is_legacy_rest()) {
+        REQUIRE(
+            ctx.data_protocol("tiledb://namespace/array") ==
```

Comment:
I know that this doesn't change the returned value of `data_protocol()`, but I would avoid namespace paths. I believe workspace/teamspace or something totally unrelated would be a better choice.